synapse/storage/data_stores/main/event_federation.py

```diff
@@ -174,23 +174,24 @@ def _get_auth_chain_difference_txn(
         }
 
         # We need to get the depth of the initial events for sorting purposes.
-        rows = self.db.simple_select_many_txn(
-            txn,
-            table="events",
-            column="event_id",
-            iterable=initial_events,
-            keyvalues={},
-            retcols=("event_id", "depth"),
+        sql = """
+            SELECT depth, event_id FROM events
+            WHERE %s
+            ORDER BY depth ASC
+        """
+        clause, args = make_in_list_sql_clause(
+            txn.database_engine, "event_id", initial_events
         )
+        txn.execute(sql % (clause,), args)
 
-        # The sorted list of events we should walk the auth chain off.
-        search = sorted((row["depth"], row["event_id"]) for row in rows)
+        # The sorted list of events whose auth chains we should walk.
+        search = txn.fetchall()
```

Comment:
probably wants a type declaration

```suggestion
        search = txn.fetchall()   # type: List[Tuple[int, str]]
```